tests/backend/test_rpc_handshake.py

```diff
@@ -0,0 +1,243 @@
+# Parsec Cloud (https://parsec.cloud) Copyright (c) AGPL-3.0 2016-present Scille SAS
+
+import pytest
+from typing import Union
+from unittest.mock import patch
+
+from parsec._parsec import DateTime, LocalDevice, DeviceID
+from parsec.api.version import ApiVersion, API_VERSION
+from parsec.serde import unpackb
+
+from tests.common import (
+    AuthenticatedRpcApiClient,
+    AnonymousRpcApiClient,
+    generate_request_signature,
+)
+
+
+async def _test_good_handshake(client: Union[AuthenticatedRpcApiClient, AnonymousRpcApiClient]):
+    # Sanity check: make sure base query is valid
+    rep = await client.send_ping(check_rep=False)
+    assert rep.status_code == 200
+    assert rep.headers["Api-Version"] == str(API_VERSION)
+
+
+async def _test_handshake_bad_organization(
+    client: Union[AuthenticatedRpcApiClient, AnonymousRpcApiClient]
+):
+    for badorg in [
+        "dummy",  # Unknown organization
+        "a" * 65,  # Invalid organization ID
+    ]:
+
+        def _before_send_hook(args):
+            args["path"] = args["path"].replace("CoolOrg", badorg)
+
+        rep = await client.send_ping(
+            before_send_hook=_before_send_hook,
+            check_rep=False,
+        )
+        assert rep.status_code == 404
+        assert rep.headers["Api-Version"] == str(API_VERSION)
+
+
+async def _test_handshake_api_version_header(
+    client: Union[AuthenticatedRpcApiClient, AnonymousRpcApiClient]
+):
+    server_versions = [ApiVersion(2, 1), ApiVersion(3, 1), ApiVersion(4, 1)]
+
+    with patch("parsec.backend.asgi.rpc.SUPPORTED_API_VERSIONS", server_versions):
+
+        # Plain invalide header value
+        rep = await client.send_ping(
+            extra_headers={"Api-Version": "dummy"},
+            check_rep=False,
+        )
+        assert rep.status_code == 422
+        assert rep.headers["Supported-Api-Versions"] == "2.1;3.1;4.1"
+
+        # Missing header, fallback to default value for backward compatibility
+        rep = await client.send_ping(
+            extra_headers={"Api-Version": None},
+            check_rep=False,
+        )
+        # Special case for anynous
+        assert rep.status_code == 200
+        rep.headers["Api-Version"] == "3.0"
+
+        # Client provides an incompatible api version
+        unknown_major = ApiVersion(version=5, revision=0)
```

Comment:
```suggestion
        unknown_major = ApiVersion(version=API_VERSION+5, revision=0)
```

Reply:
This test is independent of the actual versions supported by the server (cf. the `with patch(...)`)